driver/src/price_finding/naive_solver.rs

```diff
@@ -643,6 +663,92 @@ pub mod tests {
         assert!(!res.is_non_trivial());
     }
 
+    #[test]
+    fn test_match_and_lonely_order() {
```

Comment:
I don't really understand what this means. Can you clarify?

Reply:
There are two orders that match and then a single order that is unrelated and cannot be matched.

Reply:
I'll change lonely to unmatchable.